six/two/two.cpp

```diff
@@ -521,8 +515,14 @@ char **Two::getCheckWarnings(SixPyObject *check)
 
     for (Py_ssize_t idx = 0; idx < numWarnings; idx++) {
         PyObject *warn = PyList_GetItem(warns_list, idx); // borrowed ref
+        if (warn == NULL) {
+            PyErr_Clear();
+            setError("there was an error browsing warnings list");
+            return NULL;
```

Comment:
same: need `Py_XDECREF(warns_list);` here before returning

(might be moved to a `done:` label? Would need to be careful about the returned value of `warnings`)